src/core/load-fragment.c

```diff
@@ -2599,17 +2572,16 @@ int config_parse_unit_requires_mounts_for(
                         return 0;
                 }
 
-                if (!utf8_is_valid(word)) {
-                        log_syntax_invalid_utf8(unit, LOG_ERR, filename, line, rvalue);
-                        continue;
-                }
-
                 r = unit_full_printf(u, word, &resolved);
                 if (r < 0) {
                         log_syntax(unit, LOG_ERR, filename, line, r, "Failed to resolve unit specifiers in '%s', ignoring: %m", word);
                         continue;
                 }
 
+                r = path_simplify_and_warn(resolved, PATH_CHECK_ABSOLUTE, unit, filename, line, lvalue);
+                if (r < 0)
+                        continue;
+
```

Comment:
The check for utf8-validity needs to be before `unit_full_printf` and its error message. I don't think you can remove the utf8 check.

Reply:
I see... Are there any possibility that `unit_{full,name}_printf()` generate utf8 invalid strings? I mean, that should we check utf8-validity before _and_ after `unit_{full,name}_printf()`?

Reply:
I added utf8-validity check in `parse_line()`. So, conf parsers do not need to call `utf8_is_valid()` now.

Reply:
That's a great solution.